src/coreclr/src/gc/gc.cpp

```diff
@@ -15721,12 +15731,16 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
         else
 #endif // !FEATURE_REDHAWK
         {
+            gc_heap::gc_data_global.gen_to_condemn_reasons.set_condition(j8);
             n = max_generation;
         }
     }
 #endif //BACKGROUND_GC
 #endif //STRESS_HEAP
 
+    // TODO: Fix the 1st parameter
+    gc_heap::gc_data_global.gen_to_condemn_reasons.set_gen(gen_alloc_budget, n);
```

Comment:
>gc_heap::gc_data_global.gen_to_condemn_reasons.set_gen(gen_alloc_budget, n); [](start = 4, length = 76)

this is unnecessary since we already record the final generation we condemn in the GCStart event.